src/Device/Driver/LX_EOS/LXEosDownload.cpp

```diff
@@ -0,0 +1,270 @@
+  std::byte response[3];
+  response[0] = std::byte{ 0x06 };
+
+  auto data_array = reinterpret_cast<std::byte*>(&data);
+  std::span<std::byte> message(data_array, sizeof(data));
+
+  if (!WriteAndWaitForACK(message, env))
+    return false;
+
+  try {
+    port.FullRead({ response + 1, sizeof(response) - 1 },
+                  env,
+                  std::chrono::milliseconds(5000));
+  } catch (...) {
+    return false;
+  }
+
+  // Check CRC
+  if (CalculateCRC(response, sizeof(response), 0xFF) != 0x00)
+    return false;
+
+  flight_count = std::to_integer<uint8_t>(response[1]);
+  return true;
+}
+
+bool
+LXEosDevice::GetFlightInfo(const uint8_t index,
+                           RecordedFlightInfo& flight,
+                           OperationEnvironment& env)
+{
+  if (index == 0) // Invalid
+    return false;
+
+  EosRequestFlightInfo data;
+  data.flight_id = index;
+  std::byte response[94];
+
+  auto data_array = reinterpret_cast<std::byte*>(&data);
+  data.crc = CalculateCRC(data_array, sizeof(data) - 1, 0xFF);
+  std::span<std::byte> message(data_array, sizeof(data));
+
+  if (!WriteAndWaitForACK(message, env))
+    return false;
+  response[0] = std::byte{ 0x06 };
+
+  try {
+    port.FullRead({ response + 1, sizeof(response) - 1 },
+                  env,
+                  std::chrono::milliseconds(5000));
+  } catch (...) {
+    return false;
+  }
+
+  if (CalculateCRC(response, sizeof(response), 0xFF) != 0x00) // Check CRC
+    return false;
+
+  /* Flight ID for downloading is the index (1 = newest),
+  not the ID from this message */
+  flight.internal.lx_eos.flight_id = index;
+
+  uint32_t julianDate, time_takeoff, time_landing, file_size;
+  std::memcpy(&julianDate, &response[13], 4);
+  std::memcpy(&time_takeoff, &response[17], 4);
+  std::memcpy(&time_landing, &response[21], 4);
+  std::memcpy(&file_size, &response[89], 4);
+  flight.date = julianToDate(julianDate);
+  flight.start_time = BrokenTime::FromSecondOfDay(time_takeoff);
+  flight.end_time = BrokenTime::FromSecondOfDay(time_landing);
+  flight.internal.lx_eos.file_size = file_size;
+  return true;
+}
+
+bool
+LXEosDevice::GetFlightLogBlock(AllocatedArray<std::byte>& block,
+                               uint16_t flight_id,
+                               uint16_t block_id,
+                               OperationEnvironment& env)
+{
+  EosRequestFlightBlock data;
+  data.flight_id = flight_id;
+  data.block_id = block_id;
+  std::byte responseHeader[5];
+  responseHeader[0] = std::byte{ 0x06 };
+
+  auto data_array = reinterpret_cast<std::byte*>(&data);
+  data.crc = CalculateCRC(data_array, sizeof(data) - 1, 0xFF);
+  std::span<std::byte> message(data_array, sizeof(data));
+
+  if (!WriteAndWaitForACK(message, env))
+    return false;
+
+  try {
+    port.FullRead(std::span{ responseHeader + 1, sizeof(responseHeader) - 1 },
+                  env,
+                  std::chrono::milliseconds(5000));
+  } catch (...) {
+    return false;
```

Comment:
What's the point of catching this exception? You discard all information about the failure by converting it to`false` here.